src/simulation_lifecycle/simulation.c

```diff
@@ -89,3 +92,58 @@ int write_sim_config(const cJSON *simulation_config, char *config_json) {
     }
     return write_data_to_file(cJSON_GetStringValue(config_output_path), config_json);
 }
+
+int run_sim(const cJSON *simulation_config){
+
+    /* Get model name - e.g. 2_2_agent_sir_config */
+    cJSON *model_id = cJSON_GetObjectItemCaseSensitive(simulation_config, SIM_MODEL_ID);
+    char *model = cJSON_GetStringValue(model_id);
+
+    /* Define executable - e.g. 2_2_agent_sir_config.exe */
+    char *model_executable = concat(model, ".exe");
+
+    /* Get config output path */
+    cJSON *config = cJSON_GetObjectItemCaseSensitive(simulation_config, SIM_CONFIG_OUTPUT_PATH);
+    char *config_path = cJSON_GetStringValue(config);
+
+    /* Define a command and run simulation:
+     * The command is changing to the directory containing the .exe file.
+     * Then, the command executes the .exe file with one argument
+     * which is the scenario config json file.
+     * Example: cd ../third_party/CellDEVS_models/tutorial/bin/ ;
+     * ./2_2_agent_sir_config.exe ../../../../test/data/run_simulation/config.json
+     */
+    char *command = concat("cd ",SIM_MODEL_LIBRARY);
+    command = concat(command, " ; ./");
+    command = concat(command, model_executable);
+    command = concat(command, " ");
+    command = concat(command, config_path);
+    int sim_status = system(command);
+
+    /* Check if simulation was properly ran. */
+    if (sim_status != 0){
+        return SIM_RUN_ERROR;
+    }else{
+        /* Define results filename */
+        char *results_filename = concat(model, SIM_RESULTS_END_FILENAME);
+        char *results_filename_path = concat(SIM_RESULTS_DEFAULT_PATH,results_filename);
+
+        /* Check if results file exists */
+        if (!file_exists(results_filename_path)) {
+            return SIM_RUN_NO_RESULTS;
+        }else{
+            /* Moving the results file to predefined result output path. */
+            cJSON *result = cJSON_GetObjectItemCaseSensitive(simulation_config, SIM_RESULT_OUTPUT_PATH);
+            char *result_path = cJSON_GetStringValue(result);
```

Comment:
You must ensure that `result_path` is not `NULL`. I usually do:
```
char *result_path = cJSON_GetStringValue(cJSON_GetObjectItemCaseSensitive(simulation_config, SIM_RESULT_OUTPUT_PATH);
if (result_path == NULL) {
    return CORRESPONDING_ERROR;
}
...
```

Reply:
I did not do these checks since this function is always executed after other functions that make sure that the json file contains all the info. I guess there is no harm in adding these extra checks :)

Reply:
Actually, do we want to redo these checks? I would have to do it also for:
`cJSON *model_id = cJSON_GetObjectItemCaseSensitive(simulation_config, SIM_MODEL_ID);`
`cJSON *config = cJSON_GetObjectItemCaseSensitive(simulation_config, SIM_CONFIG_OUTPUT_PATH);`

Reply:
`I think that each function should check their corresponding parts (mine does so).
For example, the `validate_workflow` won't check the complete validity of the workflow, but it checks that at first sight, it looks OK (we can discuss this on Friday).

Reply:
Do you mean that for model_id, for example, I should verify if empty, null, does not exist, etc. and do the appropriate tests?

Reply:
Not that part, because I already checked in the `build_simulation_scenario`. But the path to the results yes (this function is the very first one that reads it)